src/app.js

```diff
@@ -72,18 +72,26 @@ const dataHandler = async (messageSet, topic, partition) => {
         const payload = await ProcessorService.processCreate(messageJSON)
         if (payload) {
           logger.info(`Sending request to scan the file ${payload.fileName}.`)
-          await producer.send({
-            topic: config.AVSCAN_TOPIC,
-            message: {
-              value: JSON.stringify({
-                topic: config.AVSCAN_TOPIC,
-                originator: 'submission-processor',
-                timestamp: new Date().toISOString(),
-                'mime-type': 'application/json',
-                payload
-              })
+          do {
+            let failed=false
+            const result = await producer.send({
+              topic: config.AVSCAN_TOPIC,
+              message: {
+                value: JSON.stringify({
+                  topic: config.AVSCAN_TOPIC,
+                  originator: 'submission-processor',
+                  timestamp: new Date().toISOString(),
+                  'mime-type': 'application/json',
+                  payload
+                })
+              }
+            })
+            const error = _.get(result, '[0].error')
+            if(error) {
+              logger.error('Raising message to scan the file failed:' + JSON.stringify(error))
+              failed = true
             }
-          })
+          } while (failed)
```

Comment:
I'm not sure I get the logic for the `do / while` loop here. 

If the `producer.send` call fails with an error, `failed` will be true, and the `while` loop will execute again. If it keeps failing with an error, it will be in an infinite loop. Am I totally misreading this?